crates/rust-analyzer/src/global_state.rs

```diff
@@ -213,7 +213,7 @@ impl GlobalStateSnapshot {
     pub(crate) fn anchored_path(&self, file_id: FileId, path: &str) -> Url {
         let mut base = self.vfs.read().0.file_path(file_id);
         base.pop();
-        let path = base.join(path);
+        let path = base.join(path).unwrap();
```

Comment:
This can only panic for virtual files, in which case the `unwrap()` below would also panic. Not sure if this is the best solution, but making `anchored_path` return `Option` means that some of the LSP methods become infallible too, so I left it at this.

Reply:
Yeah, this should be fine, we "guarantee| that each path has at least one segment (ie, paths represent only files, not directories). I guess some day we might want to add some assertions to that end, but let's punt on that for now. 

bors r+